src/server/svr_jobfunc.c

```diff
@@ -5653,8 +5655,10 @@ svr_setjob_histinfo(job *pjob, histjob_type type)
 	if ((pjob->ji_qs.ji_state != JOB_STATE_MOVED) &&
 		(pjob->ji_qs.ji_state != JOB_STATE_EXPIRED) &&
 		(pjob->ji_qs.ji_state != JOB_STATE_FINISHED)) {
-		account_entity_limit_usages(pjob, NULL, NULL, DECR);
-		account_entity_limit_usages(pjob, pjob->ji_qhdr, NULL, DECR);
+		account_entity_limit_usages(pjob, NULL, NULL, DECR,
+				pjob->ji_etlimit_decr_queued ? ETLIM_ACC_ALL_MAX : ETLIM_ACC_ALL);
+		account_entity_limit_usages(pjob, pjob->ji_qhdr, NULL, DECR,
+				pjob->ji_etlimit_decr_queued ? ETLIM_ACC_ALL_MAX : ETLIM_ACC_ALL);
```

Comment:
if `pjob->ji_etlimit_decr_queued` is set, then we now have to `DECR` only `set_entity_*_sum_max` as `set_entity_*_sum_queued` was already decremented when the job entered run. Hence passing `ETLIM_ACC_ALL_MAX`

if `pjob->ji_etlimit_decr_queued` is unset, then we have to decrement all the 4 limit attributes. Hence passing `ETLIM_ACC_ALL`